### PR TITLE
FS: Implemented access to the SD card

### LINKED ISSUE #219: pfba crash with unimplemented function and assertion failed (with unicorn)
## Hardware specs :
- Nvidia GeForce GTX 960
- AMD FX-6300
- 8 GB Of Ram
- Windows 7 Pro

## Issue :
when i run pfba with unicorn the crash reason is different than crash reason with dynarmic. Here yuzu crash after couples of seconds with "Service <Error> core/hle/service/service.cpp:ReportUnimplementedFunction:118: unknown / unimplemented function '8': port: 'ISystemClock' cmd_buf={[0]=0x10004, [1]=0xa, [2]=0x3011000, [3]=0xf7ffe328, [4]=0x49434653, [5]=0x0, [6]=0x8, [7]=0x0, [8]=0x1}" and also with assertion failed

## Log :
major part of the log because there is a bug with the command "yuzu.exe 2> log.txt" causes partial record and fully record when yuzu crash
[Click Here](https://gist.github.com/greggameplayer/70779a3473fcccaea213c95ba81900ba)

### LINKED ISSUE #247: SD Card not Implemented?
Im guessing the SDCard hasnt been implemented into the yuzu emulator? I have tested the SD card. aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaannnnnnnnnnndddddddddddddd this happend.

Here is the source code and screenshot.

[current.zip](https://github.com/yuzu-emu/yuzu/files/1823021/current.zip)
capture

### PR DESCRIPTION
**Pull request by Subv**
This also implements the IDirectory interface.
The code was tested with libtransistor's test_fs_stress program, which now runs correctly.

Closes #247
Closes #219

### CODE REVIEW

**Comment by TurtleP** [comment]
Having run this for my applet on libnx crashes (and I have what I need where I'd usually set things up) https://i.imgur.com/6LMJ7FO.png

To be honest, I've been waiting for SD access since my app requires it (we don't have romfs building yet).

Attached is my *.nro file and the folder which goes on the root of sdmc to run it. Hopefully this will allow you to test more things on the libnx side.

The crash is during startup: the app creates a folder on the SD card at `sdmc:/switch/LovePotion` and then a directory inside that for the current running game's save data (`sdmc:/switch/LovePotion/SuperGame` by default). I can't test whether file operations like open/read/write/flush/close work because of this.

[LovePotion.zip](https://github.com/yuzu-emu/yuzu/files/1829337/LovePotion.zip)

File: src/core/file_sys/disk_filesystem.cpp [hunk 1]
```diff
@@ -17,10 +17,30 @@ std::string Disk_FileSystem::GetName() const {
...
+    } else {
+        if (mode_flags & static_cast<u32>(Mode::Read))
+            mode_str = "r";
+        else if (mode_flags & static_cast<u32>(Mode::Append))
+            mode_str = "a";
+        else if (mode_flags & static_cast<u32>(Mode::Write))
+            mode_str = "w";
+    }
+
+    mode_str += "b";
```
Comment (lioncash) [inline src/core/file_sys/disk_filesystem.cpp]: The code up to this point can probably be put in its own `ModeFlagsToString` function.

Comment (bunnei) [reply src/core/file_sys/disk_filesystem.cpp]: @Subv do you plan to make this change?

File: src/core/file_sys/disk_filesystem.cpp [hunk 2]
```diff
@@ -11,16 +11,41 @@
...
+}
```
Comment (lioncash) [inline src/core/file_sys/disk_filesystem.cpp]: The string needs to be returned from the function. It's currently missing a `return mode_str;`

**Pull request merged**